contracts/0.4.24/Lido.sol

```diff
@@ -276,65 +468,104 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
         uint256 rewardBase = (appearedValidators.mul(DEPOSIT_SIZE)).add(BEACON_BALANCE_POSITION.getStorageUint256());
 
         // Save the current beacon balance and validators to
-        // calcuate rewards on the next push
+        // calculate rewards on the next push
         BEACON_BALANCE_POSITION.setStorageUint256(_beaconBalance);
         BEACON_VALIDATORS_POSITION.setStorageUint256(_beaconValidators);
 
+        // If LidoExecutionLayerRewardsVault address is not set just do as if there were no execution layer rewards at all
+        // Otherwise withdraw all rewards and put them to the buffer
+        // Thus, execution layer rewards are handled the same way as beacon rewards
+
+        uint256 executionLayerRewards;
+        address executionLayerRewardsVaultAddress = getELRewardsVault();
+
+        if (executionLayerRewardsVaultAddress != address(0)) {
+            executionLayerRewards = ILidoExecutionLayerRewardsVault(executionLayerRewardsVaultAddress).withdrawRewards(
+                (_getTotalPooledEther() * EL_REWARDS_WITHDRAWAL_LIMIT_POINTS_POSITION.getStorageUint256()) / TOTAL_BASIS_POINTS
+            );
+
+            if (executionLayerRewards != 0) {
+                BUFFERED_ETHER_POSITION.setStorageUint256(_getBufferedEther().add(executionLayerRewards));
+            }
+        }
+
+        // Don’t mint/distribute any protocol fee on the non-profitable Lido oracle report
+        // (when beacon chain balance delta is zero or negative).
+        // See ADR #3 for details: https://research.lido.fi/t/rewards-distribution-after-the-merge-architecture-decision-record/1535
         if (_beaconBalance > rewardBase) {
             uint256 rewards = _beaconBalance.sub(rewardBase);
-            distributeRewards(rewards);
+            distributeRewards(rewards.add(executionLayerRewards));
```

Comment:
I see now `distributeRewards()` as not an accurate name. The rewards distribution also happens above in line

    BEACON_BALANCE_POSITION.setStorageUint256(_beaconBalance)
   
when the total supply is increased and all the stETH balances go up.

What happens in `distributeRewards()` is taking part of the rewards as fees and distributing them to the protocol and the node operators.

Not sure about the better naming. As a hint, a naive coarse naming could be like: `takeAndDistributePartOfTheRewardsAsFees()`, `distributeProtocolAndNodeOperatorsRewards`

Reply:
I'd either leave it as is or go with smth like `distributeFees` since what it actually does is taking fees (by minting tokens). Smth like `mintFees` is even closer to the truth but might mislead a casual reader.

Reply:
Let it be `distributeFee`